core/src/apps/cardano/helpers/paths.py

```diff
@@ -18,10 +18,11 @@
 
 # the maximum allowed change address.  this should be large enough for normal
 # use and still allow to quickly brute-force the correct bip32 path
-MAX_CHANGE_ADDRESS_INDEX = const(1_000_000)
-MAX_ACCOUNT_INDEX = const(100)
+MAX_SAFE_CHANGE_ADDRESS_INDEX = const(1_000_000)
+MAX_SAFE_ACCOUNT_INDEX = const(100) | HARDENED
 ACCOUNT_PATH_INDEX = const(2)
-BIP_PATH_LENGTH = const(5)
+PURPOSE_PATH_INDEX = const(3)
```

Comment:
`purpose` in BIP-44 refers to the _first_ component.

The fourth is called `change`, which is not appropriate here. Maybe "address type" ? But that also means something else

Reply:
yes indeed, I mixed up the terminology here. The corresponding part of the path is actually called "chain" in Cardano terminology: https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/CIP-1852.md